geopandas/array.py

```diff
@@ -372,10 +379,18 @@ def geom_type(self):
 
     @property
     def area(self):
+        if self.crs and self.crs.is_geographic:
```

Comment:
I wonder if this check could get converted to an annotation `@warn_geographic` or a function `warn_geographic("area")` to consolidate the repeated logic.